src/repositories/request-repository.ts

```diff
@@ -263,21 +297,31 @@ export class RequestRepository {
     const now = new Date()
     const anchoringDeadline = new Date(now.getTime() - this.config.maxAnchoringDelayMS)
 
-    return connection.transaction(
-      async (trx) => {
-        const streamsToAnchor = await findStreamsToAnchor(trx, maxStreamLimit, now)
+    return connection
+      .transaction(
+        async (trx) => {
+          const streamsToAnchor = await findStreamsToAnchor(trx, maxStreamLimit, now)
```

Comment:
could all the logic from this line through line 325 be moved into `findStreamsToAnchor`?  Then findStreamsToAnchor could just return an array of StreamIDs, without needing to define the new `streamAndEarliestRequestDate` type, plus the logic in this function gets smaller and more straightforward